custom/icds/models.py

```diff
@@ -172,9 +180,18 @@ def update_status(self, new_status):
         HostedCCZ.objects.filter(id=self.pk).update(status=new_status)
 
 
+class HostedCCZCustomSupportingFile(models.Model):
+    link = models.ForeignKey(HostedCCZLink, on_delete=models.PROTECT)
```

Comment:
since a file can be shared by multiple links, don't delete it if the link is getting deleted.